session_subscribe_update_test.go

```diff
@@ -0,0 +1,579 @@
+		
+		// First, add a local track to simulate an active subscription
+		lt := &localTrack{
+			requestID:  42,
+			trackAlias: 100,
+		}
+		s.localTracks.addPending(lt)
+		s.localTracks.confirm(42) // Move to open state
+		
+		// Expect the message to be enqueued for the application handler
+		mh.EXPECT().enqueue(context.Background(), &Message{
+			Method:             MessageSubscribeUpdate,
+			RequestID:          42,
+			TrackAlias:         100,
+			SubscriberPriority: 50,
+			StartGroup:         10,
+			StartObject:        5,
+			EndGroup:           20,
+			Forward:            1,
+		}).Return(nil)
+		
+		// Send SUBSCRIBE_UPDATE
+		err := s.onSubscribeUpdate(&wire.SubscribeUpdateMessage{
+			RequestID:          42,
+			StartLocation:      wire.Location{Group: 10, Object: 5},
+			EndGroup:           20,
+			SubscriberPriority: 50,
+			Forward:            1,
+			Parameters:         wire.KVPList{},
+		})
+		
+		assert.NoError(t, err)
+	})
+
+	t.Run("onSubscribeUpdate_pending_subscription", func(t *testing.T) {
+		ctrl := gomock.NewController(t)
+		cms := NewMockControlMessageSendQueue[wire.ControlMessage](ctrl)
+		mh := NewMockControlMessageRecvQueue[*Message](ctrl)
+		
+		s := newSession(cms, mh, ProtocolQUIC, PerspectiveServer)
+		
+		// Add a pending local track (not yet confirmed)
+		lt := &localTrack{
+			requestID:  42,
+			trackAlias: 100,
+		}
+		s.localTracks.addPending(lt)
+		// Don't confirm it - leave it in pending state
+		
+		// Should still find the track since findByID checks both pending and open
+		mh.EXPECT().enqueue(context.Background(), gomock.Any()).Return(nil)
+		
+		err := s.onSubscribeUpdate(&wire.SubscribeUpdateMessage{
+			RequestID:          42,
+			StartLocation:      wire.Location{Group: 10, Object: 5},
+			EndGroup:           20,
+			SubscriberPriority: 50,
+			Forward:            1,
+			Parameters:         wire.KVPList{},
+		})
+		
+		assert.NoError(t, err)
+	})
+
+	t.Run("receive_subscribe_update_message", func(t *testing.T) {
+		ctrl := gomock.NewController(t)
+		cms := NewMockControlMessageSendQueue[wire.ControlMessage](ctrl)
+		mh := NewMockControlMessageRecvQueue[*Message](ctrl)
+		
+		s := newSession(cms, mh, ProtocolQUIC, PerspectiveServer)
+		
+		// Add a local track
+		lt := &localTrack{
+			requestID:  42,
+			trackAlias: 100,
+		}
+		s.localTracks.addPending(lt)
+		s.localTracks.confirm(42)
+		
+		// Mark handshake as done
+		close(s.handshakeDoneCh)
+		
+		// Expect the message to be passed to application
+		mh.EXPECT().enqueue(context.Background(), gomock.Any()).Return(nil)
+		
+		// Simulate receiving SUBSCRIBE_UPDATE via control stream
+		err := s.receive(&wire.SubscribeUpdateMessage{
+			RequestID:          42,
+			StartLocation:      wire.Location{Group: 10, Object: 5},
+			EndGroup:           20,
+			SubscriberPriority: 50,
+			Forward:            1,
+			Parameters:         wire.KVPList{},
+		})
+		
+		assert.NoError(t, err)
+	})
+
+	t.Run("subscribe_update_priority_values", func(t *testing.T) {
```

Comment:
This also seems to be the same as above, except that it tests some different priority values? I don't think we need to even test that since we don't do any validation on the priority values.